File: android/app/src/main/java/com/travelexpense/data/model/Trip.kt
package com.travelexpense.data.model

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable
import java.time.LocalDate
import java.time.LocalDateTime

/**
 * 旅行モデル
 */
@Serializable
data class Trip(
    val id: String,
    @SerialName("user_id")
    val userId: String,
    val name: String,
    val destination: String? = null,
    @SerialName("start_date")
    val startDate: String, // ISO 8601 format
    @SerialName("end_date")
    val endDate: String? = null,
    val budget: Double? = null,
    val currency: String,
    @SerialName("created_at")
    val createdAt: String
)

/**
 * 旅行作成用リクエスト
 */
@Serializable
data class CreateTripRequest(
    @SerialName("user_id")
    val userId: String,
    val name: String,
    val destination: String? = null,
    @SerialName("start_date")
    val startDate: String,
    @SerialName("end_date")
    val endDate: String? = null,
    val budget: Double? = null,
    val currency: String
)

/**
 * 旅行更新用リクエスト
 */
@Serializable
data class UpdateTripRequest(
    val name: String? = null,
    val destination: String? = null,
    @SerialName("start_date")
    val startDate: String? = null,
    @SerialName("end_date")
    val endDate: String? = null,
    val budget: Double? = null,
    val currency: String? = null
)

File: ios/TravelExpense/Models/Trip.swift
import Foundation

/// 旅行モデル
struct Trip: Codable, Identifiable {
    let id: String
    let userId: String
    let name: String
    let destination: String?
    let startDate: Date
    let endDate: Date?
    let budget: Double?
    let currency: String
    let createdAt: Date

    enum CodingKeys: String, CodingKey {
        case id
        case userId = "user_id"
        case name
        case destination
        case startDate = "start_date"
        case endDate = "end_date"
        case budget
        case currency
        case createdAt = "created_at"
    }
}

/// 旅行作成用リクエスト
struct CreateTripRequest: Codable {
    let userId: String
    let name: String
    let destination: String?
    let startDate: Date
    let endDate: Date?
    let budget: Double?
    let currency: String

    enum CodingKeys: String, CodingKey {
        case userId = "user_id"
        case name
        case destination
        case startDate = "start_date"
        case endDate = "end_date"
        case budget
        case currency
    }
}

/// 旅行更新用リクエスト
struct UpdateTripRequest: Codable {
    let name: String?
    let destination: String?
    let startDate: Date?
    let endDate: Date?
    let budget: Double?
    let currency: String?

    enum CodingKeys: String, CodingKey {
        case name
        case destination
        case startDate = "start_date"
        case endDate = "end_date"
        case budget
        case currency
    }
}
